cypress/integration/models/stakeholders.ts

```diff
@@ -22,53 +37,108 @@ export class Stakeholders {
+    }
+
+    protected selectGroups(groups: Array<string>): void {
+        groups.forEach(function (group) {
+            selectFormItems(groupInput, group);
+        });
+    }
+
+    protected removeGroups(groups: Array<string>): void {
+        groups.forEach(function (group) {
+            removeMember(group);
+        });
+    }
+
     getStakeholderName(): string {
-        this.stakeholderName = faker.name.findName();
-        return this.stakeholderName;
+        return faker.name.findName();
     }
 
     getStakeholderEmail(): string {
-        this.stakeholderEmail = faker.internet.email();
-        return this.stakeholderEmail;
+        return faker.internet.email();
     }
 
-    create(): void {
+    create({
+        name = this.stakeholderName,
+        email = this.stakeholderEmail,
+        jobfunction = null,
+        groups = [],
+        cancel = false,
+    } = {}): void {
         Stakeholders.clickStakeholders();
-        clickByText("button", "Create new");
-        this.getStakeholderName();
-        this.getStakeholderEmail();
-        this.fillEmail(this.stakeholderEmail);
-        this.fillName(this.stakeholderName);
-        submitForm();
+        clickByText(button, createNewButton);
+        if (cancel) {
+            cancelForm();
+        } else {
+            this.fillEmail(email);
+            this.fillName(name);
+            if (jobfunction) {
+                this.selectJobfunction(jobfunction);
+            }
+            if (groups.length != 0) {
+                this.selectGroups(groups);
+            }
+            submitForm();
+            checkSuccessAlert(
+                commonView.successAlertMessage,
+                `Success! ${name} was added as a stakeholder.`
+            );
+        }
     }
 
-    edit(jobFunction?: boolean, groups?: boolean): void {
+    edit({
+        name = this.stakeholderName,
+        email = this.stakeholderEmail,
+        jobfunction = null,
+        groups = [],
+        cancel = false,
+    } = {}): void {
         Stakeholders.clickStakeholders();
         selectItemsPerPage(100);
         cy.wait(2000);
         cy.get(tdTag)
             .contains(this.stakeholderEmail)
             .parent(trTag)
             .within(() => {
-                click(editButton);
+                click(commonView.editButton);
             });
-        this.getStakeholderName();
-        this.fillName(this.stakeholderName);
-        // jobfunction and group edit to be implemented
-        submitForm();
+        if (
+            !cancel &&
+            (name != this.stakeholderName ||
+                email != this.stakeholderEmail ||
+                jobfunction ||
+                groups.length != 0)
+        ) {
+            this.fillName(name);
+            if (email != this.stakeholderEmail) this.fillEmail(email);
+            if (jobfunction) this.selectJobfunction(jobfunction);
+            if (groups.length != 0) this.selectGroups(groups);
```

Comment:
@sshveta , that is because the upper condition is an OR, which just takes the code inside condition. If I will not put a double check, and re-assign values, it won't work, since suppose if in the function call, we dont pass jobfunction value, it will replace it with null without the if check.

Alternate way is to use separate if for all, but then we would need an extra flag to verify if any of the updated value was sent, then only we can submit form.

Reply:
@sshveta I agree with @nitishSr 
First `if` only checks if we have any new value to edit and second inside `if`s checks values that are optional(by default `null`) which can not be filled directly. Like he had given example of jobfunction